fsd_utils/authentication/decorators.py

```diff
@@ -52,18 +54,31 @@ def _check_access_token(auto_redirect=True):
     login_cookie = request.cookies.get(user_token_cookie_name)
     if not login_cookie:
         if auto_redirect:
-            _failed_redirect()
+            _failed_redirect(return_app)
         return False
 
     try:
         return validate_token_rs256(login_cookie)
     except (PyJWTError, ExpiredSignatureError):
         if auto_redirect:
-            _failed_redirect()
+            _failed_redirect(return_app)
         return False
 
 
-def login_required(f=None, roles_required: List[str] = None):
+def _build_logout_url(authenticator_host: str, return_app: SupportedApp | None):
```

Comment:
continuing from my comment on authenticator_host - may not need to pass this as an argument at all but instead call it in the function when needed through the common config :)

Reply:
Nice, I like this idea 👍